app/services/calculate_service.rb

```diff
@@ -26,6 +29,10 @@ def apply_discount(discount)
 
   def calculate_price_discount(item, discount)
    return item.line_item.price if item.line_item.keto?  
+   # CON: It seems like line_item.price could be nil and the
```

Comment:
I agree, I believe I could have handled the errors better and added some unit tests to ensure higher code quality.